release-deploy/action.yml

```diff
@@ -19,7 +21,11 @@ runs:
       shell: bash
     - run: |
         echo ${{ github.token }} | gh auth login --with-token
-        zipName="*-${{ inputs.gyg_env }}-${{ inputs.gyg_region }}-${{ inputs.version }}.zip"
+        zipName="*-${{ inputs.gyg_env }}-${{ inputs.gyg_region }}-${{ inputs.version }}"
+        if [[ -n "${{ inputs.branch }}" ]]; then
+          zipName="$zipName-${{ inputs.branch }}";
+        fi
+        zipName="${zipName}.zip"
```

Comment:
One thing I note that we're not doing... in the JIRA ticket, the last paragraph talks about preventing deployment to production unless the artifact matches some pre-determined pattern.

Initially, I was thinking that this would be useful to prevent people accidentally deploying the group orders artifact into production. However, now that I think about it more, it will also probably mean we can't deploy `v1.2.3-hotfix` into production 🤔 

Let's have an offline discussion about this.

Reply:
On further reflection, I think that this code is moving us forward in terms of capability. Restricting which branches can go to which environments is a secondary requirements.